models/skills.js

```diff
@@ -0,0 +1,75 @@
+const firestore = require("../utils/firestore");
+const skillsCollection = firestore.collection("skills");
+const users = require("./users");
+
+/**
+ * Award a Skill
+ *
+ * @param skillData { Object }:  Skill data object to be stored in DB
+ * @param userName  { String }:   Username of user to fetch their details.
+ * @return {Promise<skill|Object>}
+ */
+async function awardSkill(skillData, userName) {
+  try {
+    const skillSnapshot = await skillsCollection.add(skillData);
+    const userInfo = await users.fetchUser({ username: userName });
+    if (!userInfo.userExists) throw new Error("User doesn't exist");
```

Comment:
Shouldn't the user be fetched before any skills are created?
What are your thoughts on it?